pygame/mixer_music.py

```diff
@@ -204,7 +204,12 @@ def queue(filename):
 def _endmusic_callback():
     global _current_music, _queue_music, _music_pos, _music_pos_time
     if _endmusic_event is not None and sdl.SDL_WasInit(sdl.SDL_INIT_AUDIO):
-        event.post(event.Event(_endmusic_event))
+        # Pygame doesn't do the same checks for this path as in
+        # event.post, and people rely on that, so we also duplicate
+        # the logic
+        event = ffi.new('SDL_Event*')
+        event.type = _endmusic_event
+        sdl.SDL_PushEvent(event)
```

Comment:
Pygame skips the SDL_NUMEVENTS checks for the end of sound callback, and the end of music callback - I haven't found any other places.

The sound case isn't identical, since it also tacks on some extra data onto the event in some circumstances, so unifying these cases isn't easy, and the benefits aren't particularly great.

I'd also rather not add this to event.post directly, since it is part of the public api.